### PR TITLE
[14.0][IMP] cetmix_tower_yaml: Server templates and keys

### PR DESCRIPTION
**Pull request by ivs-cetmix**
Implemented YAML support for Server Templates, Server Logs and Keys.

Task: 4068

## Release Notes

- **New Features**
	- Updated URLs in documentation to point to the stable version `14.0`.
	- Introduced new views for handling YAML-related fields in server templates and plans.
	- Added functionality for securely masking sensitive information in YAML outputs, including SSH passwords and secret values.
	- New model classes for enhanced logging and server template management.

- **Bug Fixes**
	- Improved clarity and consistency in the README and other documentation files.

- **Tests**
	- Added new test classes to verify the functionality of masking sensitive data in YAML outputs.

- **Chores**
	- Updated manifest and import statements to include new data files and modules.

### CODE REVIEW

File: cetmix_tower_yaml/tests/test_server_key.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+class TestTowerKey(TransactionCase):
+    def setUp(self, *args, **kwargs):
+        super().setUp(*args, **kwargs)
+
+        self.Key = self.env["cx.tower.key"]
+
+    def test_key_secret_value_mask(self):
+        """Test that secret value is masked in YAML"""
+        key = self.Key.create(
+            {"name": "Test Key", "key_type": "s", "secret_value": "test_secret"}
```
Comment (tendil) [inline cetmix_tower_yaml/tests/test_server_key.py:13]: For better readability and adherence to Odoo coding style standards, it's recommended to format dictionaries with long keys and values in a column layout. This makes the code easier to read and maintain. For example:

```python
key = self.Key.create({
    "name": "Test Key",
    "key_type": "s",
    "secret_value": "test_secret"
})
```

This format is helpful when adding new keys or modifying existing values.

File: cetmix_tower_yaml/tests/test_server_template.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+class TestTowerServerTemplate(TransactionCase):
+    def setUp(self, *args, **kwargs):
+        super().setUp(*args, **kwargs)
+
+        self.ServerTemplate = self.env["cx.tower.server.template"]
+
+    def test_server_template_ssh_password_mask(self):
+        """Test that SSH password is masked in YAML"""
+        server_template = self.ServerTemplate.create(
```
Comment (tendil) [inline cetmix_tower_yaml/tests/test_server_template.py:12]: It is recommended to add exception handling when creating the `ServerTemplate`. This will make the test more robust and help diagnose issues more easily.

File: cetmix_tower_yaml/tests/test_server_template.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+            {"name": "Test Server Template", "ssh_password": "test_password"}
+        )
+        record_dict = server_template._prepare_record_for_yaml()
+        self.assertEqual(record_dict["ssh_password"], "********")
```
Comment (tendil) [inline cetmix_tower_yaml/tests/test_server_template.py]: Add more context to the `assertEqual` message so that, in case of failure, it's clear what value was returned instead of the expected one. For example:
```python
self.assertEqual(
    record_dict.get("ssh_password"),
    "********",
    "Expected `ssh_password` to be masked as '********', but found: {}".format(record_dict.get("ssh_password"))
)
```

File: cetmix_tower_yaml/tests/test_server_template.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+from odoo.tests import TransactionCase
+
+
+class TestTowerServerTemplate(TransactionCase):
+    def setUp(self, *args, **kwargs):
+        super().setUp(*args, **kwargs)
+
+        self.ServerTemplate = self.env["cx.tower.server.template"]
+
+    def test_server_template_ssh_password_mask(self):
```
Comment (tendil) [inline cetmix_tower_yaml/tests/test_server_template.py:10]: Add additional tests to cover edge cases, such as missing `ssh_password` or using an empty value.

**Review by tendil — COMMENTED** [review]
Overall impression of the changes is positive: the code is functional and meets the key requirements of the task. However, there are a few areas for improvement:

1. Readability improvement: It’s recommended to format dictionaries with long keys and values in a column layout for easier maintenance and clarity.
2. Error handling: The current tests do not account for potential exceptions during record creation or method calls. Adding exception handling will make the tests more robust and simplify issue diagnosis.
3. Better error messages: It’s suggested to provide more detailed messages in `assertEqual` to make test failures clearer, highlighting mismatched values.
4. Edge case coverage: Adding tests for edge cases (e.g., empty or missing data) would strengthen the reliability of the implementation.

The code can be improved in terms of resilience, readability, and test coverage, but overall, it’s heading in the right direction. Great job!

File: cetmix_tower_yaml/tests/test_server_key.py (hunk 1 above)
Comment (ivs-cetmix) [reply cetmix_tower_yaml/tests/test_server_key.py:13]: I agree, however this code was automatically formatted using pre-commit scripts.

File: cetmix_tower_yaml/tests/test_server_template.py (hunk 2 above)
Comment (ivs-cetmix) [reply cetmix_tower_yaml/tests/test_server_template.py:12]: Good notice, thank you! However there is already  an exception handler [implemented](https://github.com/cetmix/cetmix-tower/blob/6c43da83327f3d6c917419759c445bbba888e2d5/cetmix_tower_yaml/models/cx_tower_yaml_mixin.py#L84) at the mixin level.
Which is being tested [here](https://github.com/cetmix/cetmix-tower/blob/6c43da83327f3d6c917419759c445bbba888e2d5/cetmix_tower_yaml/tests/test_tower_yaml_mixin.py#L411)

File: cetmix_tower_yaml/tests/test_server_template.py (hunk 3 above)
Comment (ivs-cetmix) [reply cetmix_tower_yaml/tests/test_server_template.py]: Good notice! FYI: f-strings are preferred over the .format because in most cases they are faster.

File: cetmix_tower_yaml/tests/test_server_template.py (hunk 4 above)
Comment (ivs-cetmix) [reply cetmix_tower_yaml/tests/test_server_template.py:10]: SSH password is an optional field so it should'n be an issue. However the idea to add additional tests looks absolutely reasonable.

**Review by tendil — APPROVED** [review]
Good job, I have no further comments.

**Comment by CetmixGitDrone** [comment]
What a great day to merge this nice PR. Let's do it!
Prepared branch [14.0-dev-ocabot-merge-pr-143-by-ivs-cetmix-bump-nobump](https://github.com/cetmix/cetmix-tower/commits/14.0-dev-ocabot-merge-pr-143-by-ivs-cetmix-bump-nobump), awaiting test results.

**Comment by CetmixGitDrone** [comment]
Congratulations, your PR was merged at d10dbe58742a99b65ba1454a0527fbc064e53887. Thanks a lot for contributing to cetmix. ❤️
